internal/proto/reader_test.go

```diff
@@ -36,6 +41,26 @@ var _ = Describe("Reader", func() {
 		Expect(string(data)).To(Equal("hello"))
 	})
 
+	It("should read line (overflowing the buffer)", func() {
+
+		/*
+			Generate a string which doesn't fit in the bufio internal default
+			buffer
+		*/
+		bytes := make([]byte, bufioDefaultBufSize+1)
+		for i := range bytes {
+			bytes[i] = 'x'
+		}
+
+		str := string(bytes)
+
+		p := proto.NewReader(strings.NewReader(str + "\r\n"))
+
+		data, err := p.ReadLine()
```

Comment:
Note that even if the string is split over multiple `Reader.ReadLine` calls, it's still only one line since there is not line breaker. I would expect the `p.ReadLine()` to return the entire string at once.